### ISSUE TITLE
Set default name for settings file as `project.scala`

### ISSUE THREAD

**Issue by tgodzik** [VirtusLab/scala-cli#1489]
**Describe what needs to be done and why**
In case of multiple files it might be useful to allow the user to specify all the using directives in a central file as one would expect similarly to build tools.

There was multiple discussions about what the name should and it seems the winning vote was on `project.scala`

We need to probably do a number of things:
- [x] warn users about multiple using directives and suggest using `project.scala`
- [x] automatically detect project as ScalaCLI in Metals when `project.scala` is present
- [ ] ~create a g8 template with `project.scala` and also add it to new project wizard in Metals~ (now tracked under #2108)
- [ ] ~add an automatic action to move all directives into `project.scala`~ (now tracked under #2104)
- [ ] ~fix tapir templates to use `project.scala`~ (now tracked under #2109)
- [x] update all existing references in our docs from `project.settings.scala` to `project.scala`

This issue can act as a central point coordinate the work being done.

Nice to have:
-  ~automatically detect project as ScalaCLI in Intellij when `project.scala` is present~ (now tracked under #2111)
- [ ] ~the existing docs on `project.settings.scala` seem to have been confusing to some users, so probably improve the docs a bit~ (now tracked under #2110)
  - [ ] ~we probably need a dedicated, small doc on this in the `Guides` section~
  - [ ] ~the [Project root directory doc](https://scala-cli.virtuslab.org/docs/reference/root-dir) should link to the aforementioned guide~
- [ ] ~Once all else is done, we probably should enable migrating directives to `project.scala` automatically with a sub-command, as [our docs are implying we've been planning to do that for a long while](https://scala-cli.virtuslab.org/docs/guides/using-directives#using-directives-in-the-scala-cli).~ (now tracked under #2104)

**Comment by tgodzik** [comment]
Anything is missing?

**Comment by Gedochao** [comment]
- [ ] update all existing references in our docs from `project.settings.scala` to `project.scala`

Nice to have
- [ ] the existing docs on `project.settings.scala` seem to have been confusing to some users, so probably improve the docs a bit
  - [ ] we probably need a dedicated, small doc on this in the `Guides` section
  - [ ] the [Project root directory doc](https://scala-cli.virtuslab.org/docs/reference/root-dir) should link to the aforementioned guide

Another nice to have: (this should be a separate ticket)
- [ ] Once all else is done, we probably should enable migrating directives to `project.scala` automatically with a sub-command, as [our docs are implying we've been planning to do that for a long while](https://scala-cli.virtuslab.org/docs/guides/using-directives#using-directives-in-the-scala-cli).

**Comment by tgodzik** [comment]
Thanks! Added to the description.

**Comment by tusharmath** [comment]
I feel using a dot json file is much better. As soon as one adds .scala file there is a tendency for the community to add weird DSLs potentially going in the direction of sbt. 

Json is really nice in terms of being beginner friendly. Its perfect also because you can't just add logic and tasks that easily hence constraining from becoming a build tool. If Scala CLI intends to be a productive CLI tool then it should be consistent with other cli tools, almost all of them use dot files.

For eg :
```
scala-cli --config .scalaconf Main.scala
```
In my humble opinion, this is the best dev experience, because it gets me started immediately. 
I think we can take cues from package.json files, while working with node.js. They are concise and have almost a flat learning curve. 


Let me know what you think 🙏

**Comment by tgodzik** [comment]
@tusharmath Thanks for the suggestion!

The plan currently is to only allow using directives a.k.a `//> using... ` and project.scala just allows users to specify everything in a well defined single place without forcing everyone to use it. If we choose to enable json configuration that would mean that we would have 3 sources of truth: CLI args, using directives and json configuration, which will be very difficult to maintain and make sure that works well out of the box for most people. This is not a direction we want to take, but really thank you for your input! It's really good to have a discussion about the finer points of ScalaCLI design decisions.

**Comment by Gedochao** [comment]
Extracted the remaining points to separate tickets, should be tracked there now.
Added links in this issue's description as well.
Closing this.
